File: app/src/main/java/com/example/android/moviesremake/details/DetailActivity.java
package com.example.android.moviesremake.details;

import android.content.ActivityNotFoundException;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.support.v4.app.NavUtils;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.RecyclerView;
import android.view.MenuItem;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import com.example.android.moviesremake.R;
import com.example.android.moviesremake.realm.MovieRealm;
import com.example.android.moviesremake.retrofit.model.MovieRetrofit;
import com.example.android.moviesremake.utils.HelperRxClass;
import com.squareup.picasso.Picasso;

import java.util.UUID;

import butterknife.BindView;
import butterknife.ButterKnife;
import io.realm.Realm;

import static com.example.android.moviesremake.R.id.reviews;
import static com.example.android.moviesremake.R.id.trailers;

public class DetailActivity extends AppCompatActivity implements DetailAdapter.ForecastAdapterOnClickHandler{

    private MovieRetrofit movie;
    private DetailAdapter mAdapter;
    private RecyclerView recycler;
    private RecyclerView recycler2;
    private Realm realm;


    @BindView(R.id.overview) TextView textOverview;
    @BindView(R.id.vote) TextView textVote;
    @BindView(R.id.release) TextView textRelease;
    @BindView(R.id.imageView1) ImageView imageView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_detail_relative);
        this.getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        ButterKnife.bind(this);

        Intent intentThatStartedThisActivity = getIntent();
        if (intentThatStartedThisActivity != null) {
            if (intentThatStartedThisActivity.hasExtra("movies")) {
                movie = intentThatStartedThisActivity.getParcelableExtra("movies");
                setTitle(movie.getTitle());
                Picasso.with(this).load(movie.getImage()).into(imageView);
                textRelease.setText(movie.getRelease());
                textVote.setText(movie.getVote()+"/10".toString());
                textOverview.setText(movie.getOverview());
            }
        }

        realm = Realm.getDefaultInstance();

        //http://api.themoviedb.org/3/movie/346364/reviews?api_key=c88f3eabe09958ae472c9cd7e20b38aa


        recycler = (RecyclerView) findViewById(reviews);
        recycler2 = (RecyclerView) findViewById(trailers);
        HelperRxClass hp = new HelperRxClass(movie.getId(), this, mAdapter, recycler, recycler2, this);
        hp.setReviews();

    }


    @Override
    public boolean onOptionsItemSelected(MenuItem item) {

        switch (item.getItemId()) {
            // Respond to the action bar's Back button
            case android.R.id.home:
                NavUtils.navigateUpFromSameTask(this);
                return true;
        }
        return super.onOptionsItemSelected(item);
    }

// maybe code an delete from database after clicking on it again?
    public void insertMovieToDatabase(View view){

        realm.executeTransactionAsync(new Realm.Transaction() {
            @Override
            public void execute(Realm realm) {
                MovieRealm task = realm.where(MovieRealm.class).equalTo("image", movie.getImage()).findFirst();
                if(task == null){
                    task = realm.createObject(MovieRealm.class,  UUID.randomUUID().toString());
                    task.setImage(movie.getImage());
                    task.setMovieID(movie.getId());
                    task.setOverview(movie.getOverview());
                    task.setRelease(movie.getRelease());
                    task.setVote(movie.getVote());
                }else{
                    runOnUiThread(new Runnable() {
                        public void run() {
                            Toast.makeText(getApplicationContext(), "Already in favorites!", Toast.LENGTH_SHORT).show();
                        }
                    });
                }
            }
        });

    }


    @Override
    protected void onDestroy() {
        super.onDestroy();
        realm.close();
    }


    @Override
    public void onClick(String url) {
        Intent intent = new Intent(Intent.ACTION_VIEW);
        intent.setData(Uri.parse("https://www.youtube.com/watch?v="+url));

        // Always use string resources for UI text. This says something like "Share this photo with"
        String title = (String) getResources().getText(R.string.chooser_title);
        // Create and start the chooser
        Intent chooser = Intent.createChooser(intent, title);

        try {
            this.startActivity(chooser);
        } catch (ActivityNotFoundException ex) {
            intent.setPackage("com.android.chrome");
            this.startActivity(chooser);
        }
    }
}

File: app/src/main/java/com/example/android/moviesremake/details/DetailActivity.kt
package com.example.android.moviesremake.details

import android.content.ActivityNotFoundException
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.support.v4.app.NavUtils
import android.support.v7.app.AppCompatActivity
import android.view.MenuItem
import android.view.View
import android.widget.Toast
import com.example.android.moviesremake.R
import com.example.android.moviesremake.realm.MovieRealm
import com.example.android.moviesremake.retrofit.model.MovieRetrofit
import com.example.android.moviesremake.utils.HelperRxClass
import com.squareup.picasso.Picasso
import io.realm.Realm
import kotlinx.android.synthetic.main.activity_detail_relative.*
import java.util.*

class DetailActivity : AppCompatActivity(), DetailAdapter.ForecastAdapterOnClickHandler {

    private lateinit var movie: MovieRetrofit
    private val mAdapter: DetailAdapter? = null
    private var realm: Realm? = null


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_detail_relative)
        this.supportActionBar!!.setDisplayHomeAsUpEnabled(true)

        val intentThatStartedThisActivity = intent
        if (intentThatStartedThisActivity != null) {
            if (intentThatStartedThisActivity.hasExtra("movies")) {
                movie = intentThatStartedThisActivity.getParcelableExtra("movies")
                title = movie.title
                Picasso.with(this).load(movie.getImage()).into(imageView1)
                release.text = movie.release
                vote.text = movie.vote + "/10"
                overview.text = movie.overview
            }
        }

        realm = Realm.getDefaultInstance()

        //http://api.themoviedb.org/3/movie/346364/reviews?api_key=c88f3eabe09958ae472c9cd7e20b38aa


        val hp = HelperRxClass(movie.id, this, mAdapter, reviews, trailers, this)
        hp.setReviews()

    }


    override fun onOptionsItemSelected(item: MenuItem): Boolean {

        when (item.itemId) {
        // Respond to the action bar's Back button
            android.R.id.home -> {
                NavUtils.navigateUpFromSameTask(this)
                return true
            }
        }
        return super.onOptionsItemSelected(item)
    }

    // maybe code an delete from database after clicking on it again?
    fun insertMovieToDatabase(view: View) {

        realm!!.executeTransactionAsync { realm ->
            var task: MovieRealm? = realm.where(MovieRealm::class.java).equalTo("image", movie.getImage()).findFirst()
            if (task == null) {
                task = realm.createObject(MovieRealm::class.java, UUID.randomUUID().toString())
                task.image = movie.getImage()
                task.movieID = movie.id
                task.overview = movie.overview
                task.release = movie.release
                task.vote = movie.vote
            } else {
                runOnUiThread { Toast.makeText(applicationContext, "Already in favorites!", Toast.LENGTH_SHORT).show() }
            }
        }

    }


    override fun onDestroy() {
        super.onDestroy()
        realm!!.close()
    }


    override fun onClick(trailer: String?) {
        val intent = Intent(Intent.ACTION_VIEW)
        intent.data = Uri.parse("https://www.youtube.com/watch?v=" + trailer)

        // Always use string resources for UI text. This says something like "Share this photo with"
        val title = resources.getText(R.string.chooser_title) as String
        // Create and start the chooser
        val chooser = Intent.createChooser(intent, title)

        try {
            this.startActivity(chooser)
        } catch (ex: ActivityNotFoundException) {
            intent.`package` = "com.android.chrome"
            this.startActivity(chooser)
        }

    }
}
